### PR TITLE
Update Schema Definitions and Documentation to support the Satisfaction Scale

### PR DESCRIPTION
**Pull request by VirajP1002**
### What is the context of this PR?
The aim of this PR is to update the schema definitions to support a new format for feedback and a new submission payload type, `user_research` in order to facilitate the Satisfaction Scale when it's added to Runner.

The format for the new type (named `user_research_response`) will contain `full_name` and `email address`. The updated feedback_response will contain `feedback_rating` and `feedback_text`

The changes made are the new/updated formats for `user_research_response` and `feedback_response` respectively. The current implementation for `feedback_response` is now `feedback_response_legacy`. The example schemas were updated to show the new structure of the submission payload and relevant documentation were updated to incorporate the new changes.

### How to Review
- Ensure changes made are suitable
- Check if any mentions for `user_research` are missing
- Test new JSON schemas pass validation

### Checklist

* [ ] Changes to the spec have been added to example schemas in [examples/](examples/)
* [ ] JSON Schema definitions have been updated

### CODE REVIEW

**Comment by VirajP1002** [comment]
Currently failing as new changes cause validation to fail for v1 legacy files, this will be addressed and fixed by resolving https://github.com/ONSdigital/ons-schema-definitions/pull/89

File: schemas/common/response_data.json [hunk 1]
```diff
@@ -312,30 +312,39 @@
...
+    "unevaluatedProperties": false,
+    "required": ["feedback_rating"]
+  },
+  "user_research_response": {
+    "type": "object",
+    "description": "This is used for data_version 0.0.1 or 0.0.3 user research submissions",
+    "properties": {
+      "full_name": {
+        "$ref": "definitions.json#/non_empty_string",
+        "description": "The full name of the feedback respondent."
```
Comment (berroar) [inline schemas/common/response_data.json:338]: Only minor but these relate to user research rather than feedback so may be worth updating the descriptions

**Review by berroar — APPROVED** [review]

File: schemas/common/response_data.json [hunk 2]
```diff
@@ -290,9 +290,9 @@
       "minProperties": 1
     }
   },
-  "feedback_response": {
+  "feedback_response_legacy": {
     "type": "object",
-    "description": "This is used for data_version 0.0.1 or 0.0.3 feedback submissions.",
+    "description": "This is used for data_version 0.0.1 or 0.0.3 feedback submissions by legacy v2 runner..",
```
Comment (berroar) [inline schemas/common/response_data.json:295]: I don't think the "new" legacy feedback was used by v2

Comment (MebinAbraham) [reply schemas/common/response_data.json:295]: The "new" legacy is the current one in live hence v2 uses it, unless you meant something different?.

Comment (berroar) [reply schemas/common/response_data.json:295]: I thought the currrent runner was v3 runner?

Comment (MebinAbraham) [reply schemas/common/response_data.json:295]: There is no 'v2/v3' anymore, just 1 runner, I think the wording is just misleading and unneeded.

Comment (VirajP1002) [reply schemas/common/response_data.json]: Yeah, that makes sense, the wording is incorrect and misleading, updated 👍

**Review by Jon-Kent — APPROVED** [review]

**Pull request merged**